substrate/frame/support/procedural/src/runtime/parse/mod.rs

```diff
@@ -39,6 +39,7 @@ mod keyword {
 	custom_keyword!(pallet_index);
 	custom_keyword!(disable_call);
 	custom_keyword!(disable_unsigned);
+	custom_keyword!(disable_metadata);
```

Comment:
I don't think we need this. Without metadata you can not do anything.

Reply:
> I don't think we need this. Without metadata you can not do anything.

Thanks - I  actually needed this for the test to ignore `frame_system`'s metadata. What do you suggest to be a good alternative?

Reply:
Thanks but this doesn't work. I think the type registries generate an index in the metadata which is incorrect if I just ignore the system pallet and try to match.

> Filter out the system metadata when asserting the metadata:
> 
> ```diff
> diff --git a/substrate/frame/support/test/tests/pallet.rs b/substrate/frame/support/test/tests/pallet.rs
> index 68efabce75e..7906a890593 100644
> --- a/substrate/frame/support/test/tests/pallet.rs
> +++ b/substrate/frame/support/test/tests/pallet.rs
> @@ -816,7 +816,6 @@ mod runtime {
>         pub struct Runtime;
>  
>         #[runtime::pallet_index(0)]
> -       #[runtime::disable_metadata]
>         pub type System = frame_system;
>  
>         #[runtime::pallet_index(1)]
> @@ -1954,7 +1953,7 @@ fn metadata() {
>                 _ => panic!("metadata has been bumped, test needs to be updated"),
>         };
>  
> -       pretty_assertions::assert_eq!(actual_metadata.pallets, expected_metadata.pallets);
> +       pretty_assertions::assert_eq!(actual_metadata.pallets[1..], expected_metadata.pallets);
>  }
>  
>  #[test]
> @@ -2569,8 +2568,8 @@ fn test_error_feature_parsing() {
>  fn pallet_metadata() {
>         use sp_metadata_ir::{DeprecationInfoIR, DeprecationStatusIR};
>         let pallets = Runtime::metadata_ir().pallets;
> -       let example = pallets[0].clone();
> -       let example2 = pallets[1].clone();
> +       let example = pallets[1].clone();
> +       let example2 = pallets[2].clone();
>         {
>                 // Example2 pallet is deprecated
>                 assert_eq!(
> ```

Reply:
Yes you can update the type registries with the resulting ones, we don't really assert against the type registry

Reply:
> Yes you can update the type registries with the resulting ones, we don't really assert against the type registry

Yeah, I don't think that it helps to change it once the metadata has been generated. However, I can see another way to remove this from metadata without adding `disable_metadata` attribute. Will add that and see if that works for you.

Reply:
Added. @gui1117 Could you please take a look now? Thanks